docs/decisions/0039-template-engine.md

```diff
@@ -0,0 +1,74 @@
+---
+nav_order: 39
+parent: Decision Records
+---
+
+# Template engine
+
+## Context and Problem Statement
+
+We need to choose a template engine for [custom export filters](https://docs.jabref.org/collaborative-work/export/customexports) and [AI features](https://github.com/JabRef/jabref/pull/11884).
+
+A discussion of template engines was also in [one of the JabRef repos](https://github.com/koppor/jabref/issues/392).
+
+A discussion was raised on StackOverflow ["Velocity vs. FreeMarker vs. Thymeleaf"](https://stackoverflow.com/q/1459426/10037342).
+
+## Decision Drivers
+
+* It should support different formats, in case JabRef changes in the future. Also, AI features require just plain text manipulation.
+* It should have all necessary constructs from programming languages: loops, conditions, variables. It should support list or object (map) variables.
```

Comment:
Why this? I would like to have it cooking coffee for me 😅.  "necessary constructs" is too vague. Enumarate the constructs and say why (!). You can say that this is required by the AI feature. This is OK.

Reply:
I meant, that we don't need a template engine that just does find-and-replace, but something more powerfull

Reply:
Though, any template engine must be more powerful, otherwise, it's not a template engine

Reply:
Maybe, rephrase to

* It has to offer loops to enable iterating on a list of entries.

I am very conservative here. My thought is: Is the decision driver measurable somehow. Moreover, "turing compeleness" can be achieved with very simple constructs. Therefore, one needs to be exact in the wishes.

Reply:
Maybe I'll remove it